source/nodejs/adaptivecards-templating/src/template-engine.ts

```diff
@@ -413,8 +413,17 @@ export class Template {
                     if (!evaluationResult.error) {
                         whenValue = typeof evaluationResult.value === "boolean" && evaluationResult.value;
                     }
+                    
+                    if (!evaluationResult.value) {
+                        // Value was not found, and we should warn the client that the Expression was invalid
+                        console.warn(`WARN: Unable to parse the Adaptive Expression ${when}. The $when condition has been set to false by default.`);
```

Comment:
Don't use console to report error. dev can't see it

Reply:
That's a good call out. Is there another way to warn developers without crashing templating?

Reply:
It doesn't look like either templating library currently exposes warnings/errors to callers. I think the right approach would be similar to how things are done in Adaptive Cards more broadly where we pass along a list of strings that we can add warnings to... and then make those strings available to the caller in the results we hand back. We'd obviously need to introduce that without making it a breaking change.

So for C#, I'd expect:
* add optional out parameters to the `AdaptiveCardTemplate.Expand` methods that's an array of warning strings

and for JS I'd expect:
* add an overload for the `Template.expand` method that takes an out parameter that's an array of warning strings (or alternatively, maybe an optional out parameter?)

if you think these changes fit nicely within this PR, feel free to make them. if you'd rather address them separately or think it's too much of a change to deal with right now, feel free to file a bug and reference that bug in code comments for this PR

Reply:
I've created a new issue for this here: https://github.com/microsoft/AdaptiveCards/issues/7433

Reply:
it would be nice if the addition of warning is addressed before we make next release.